main.rb

```diff
@@ -0,0 +1,93 @@
+
+module Enumerable
+    def my_each
+        for item in self
+            yield(item)
+        end
+    end
+
+  def my_each_with_index
+      for item_index in (0...self.size)
+            yield(self[item_index], item_index)
+      end
+    end
+
+  def my_select
+        new_items = []
+        my_each do |item|
+          new_items.push(yield(item))
+        end
+        new_items
+  end
+
+  def my_all?
+    items_exist = true
+      my_each do |item|
+            friends_exist = false if !yield(item)
+      end
+  items_exist
+
+end
+
+    def my_any?
+        any_item = false
+        my_each do |item|
+            any_item = true if yield(item)
+        end
+        any_item
+    end
```

Comment:
- [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- Your method doesn't consider the possibility when no block or argument is given. It should return true if the enumerable contains at least one truthy value.

- Your method doesn't consider the possibility when a class is given as an argument `(some_enumerable.my_any?(String)).`

- Your method doesn't consider the possibility when a Regex is given as an argument `(some_enumerable.my_any?(/w/)).`

- Your method doesn't consider the possibility when a string is given as an argument `(some_enumerable.my_any?('cow')).`